benchmark/src/main/java/com/datatorrent/benchmark/spillable/SpillableTestOperator.java

```diff
@@ -0,0 +1,181 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.apex.malhar.lib.state.spillable.SpillableArrayListImpl;
+import org.apache.apex.malhar.lib.state.spillable.SpillableByteArrayListMultimapImpl;
+import org.apache.apex.malhar.lib.state.spillable.SpillableByteMapImpl;
+import org.apache.apex.malhar.lib.state.spillable.SpillableStateStore;
+import org.apache.apex.malhar.lib.state.spillable.managed.ManagedStateSpillableStateStore;
+import org.apache.apex.malhar.lib.utils.serde.SerdeLongSlice;
+import org.apache.apex.malhar.lib.utils.serde.SerdeStringSlice;
+
+import com.datatorrent.api.Context.OperatorContext;
+import com.datatorrent.api.DefaultInputPort;
+import com.datatorrent.api.Operator;
+import com.datatorrent.common.util.BaseOperator;
+
+public class SpillableTestOperator extends BaseOperator implements Operator.CheckpointNotificationListener
+{
+  private static final Logger logger = LoggerFactory.getLogger(SpillableTestOperator.class);
+  
+  public static final byte[] ID1 = new byte[] {(byte)1};
+  public static final byte[] ID2 = new byte[] {(byte)2};
+  public static final byte[] ID3 = new byte[] {(byte)3};
+  
+  public SpillableByteArrayListMultimapImpl<String, String> multiMap;
+  
+  public ManagedStateSpillableStateStore store;
+  
+  public long totalCount = 0;
+  public transient long countInWindow;
+  public long minWinId = -1;
+  public long committedWinId = -1;
+  public long windowId;
+  
+  public SpillableByteMapImpl<Long, Long> windowToCount;
+  
+  public long shutdownCount = -1;
+  
+  public final transient DefaultInputPort<String> input = new DefaultInputPort<String>()
+  {
+    @Override
+    public void process(String tuple)
+    {
+      processTuple(tuple);
+    }
+  };
+      
+  public void processTuple(String tuple)
+  {
+    if (++totalCount == shutdownCount) {
+      throw new RuntimeException("Test recovery. count = " + totalCount);
+    }
+    countInWindow++;
+    multiMap.put("" + windowId, tuple);
+  }
+  
+  @Override
+  public void setup(OperatorContext context)
+  {
+    super.setup(context);
+    if (windowToCount == null) {
+      windowToCount = createWindowToCountMap(store);
+    }
+    if (multiMap == null) {
+      multiMap = createMultimap(store);
+    }
+
+    store.setup(context);
+    multiMap.setup(context);
+
+    checkData();
+  }
+
+  public void checkData()
+  {
+    long startTime = System.currentTimeMillis();
+    logger.debug("check data: totalCount: {}; minWinId: {}; committedWinId: {}; curWinId: {}", totalCount,
+        this.minWinId, committedWinId, this.windowId);
+    for (long winId = Math.max(committedWinId + 1, minWinId); winId < this.windowId; ++winId) {
+      Long count = this.windowToCount.get(winId);
+      SpillableArrayListImpl<String> datas = (SpillableArrayListImpl<String>)multiMap.get("" + winId);
+      if ((datas == null && count != null) || (datas != null && count == null)) {
+        logger.error("datas: {}; count: {}", datas, count);
```

Comment:
It should probably propagate the error (maybe with stack trace) to the test driver which should then fail the test?